js/app/src/Index.svelte

```diff
@@ -94,6 +94,19 @@
 	export let worker_proxy: WorkerProxy | undefined = undefined;
 	if (worker_proxy) {
 		setWorkerProxyContext(worker_proxy);
+
+		worker_proxy.addEventListener("progress-update", (event) => {
+			loading_text = (event as CustomEvent).detail + "...";
+		});
+		worker_proxy.addEventListener("initialization-error", (event) => {
+			const error: Error = (event as CustomEvent).detail;
+			status = {
```

Comment:
I used this error state management code which seems to be originally designed for Spaces.
Is it OK?

Reply:
I think that's fine. We can maybe figure out a way to make this more generic later.